src/components/app/app.tsx

```diff
@@ -84,13 +94,35 @@ const EventVisualizer: FC = () => {
                 {isVisualizationEmpty(currentVis) && !isVisualizationLoading ? (
                     <StartScreen />
                 ) : (
-                    <PluginWrapper
-                        isVisualizationLoading={isVisualizationLoading}
-                        visualization={currentVis}
-                        displayProperty={currentUser.settings.displayProperty}
-                        onDataSorted={onDataSorted}
-                        onResponseReceived={onResponseReceived}
-                    />
+                    <>
+                        <PluginWrapper
+                            isVisualizationLoading={isVisualizationLoading}
+                            visualization={currentVis}
+                            displayProperty={
+                                currentUser.settings.displayProperty
+                            }
+                            onDataSorted={onDataSorted}
+                            onResponsesReceived={onResponsesReceived}
+                        />
+                        {interpretationId && (
+                            <InterpretationModal
+                                interpretationId={interpretationId}
+                                visualization={savedVis}
+                                isVisualizationLoading={isVisualizationLoading}
+                                pluginComponent={PluginWrapper}
+                                downloadMenuComponent={ModalDownloadDropdown}
+                                onClose={() => {
+                                    dispatch(
+                                        setNavigationState({
+                                            visualizationId: savedVis.id,
+                                            interpretationId: undefined,
+                                        })
+                                    )
+                                }}
+                                onResponsesReceived={onResponsesReceived}
+                            />
+                        )}
```

Comment:
I feel like this JSX plus all the selectors/callbacks/thunks it uses should become a dedicated component

Reply:
The `InterpretationModal` has now been extracted in a separate component in the app which renders the analytics modal component.
At the same time a provider has been added to handle `initialFocus` internally without needing a query string argument in the URL or something in the Redux state.